DynamicSmoothedGUI.py

```diff
@@ -96,57 +101,84 @@ def create_widgets(self, empty=True):
-                else:
-                    if (
-                            curr_tile.isObstacle == False and curr_tile.isBloated == False and curr_tile not in self.visitedSet):
-                        self.canvas.itemconfig(
-                            curr_rec, outline="#545454", fill="#545454")
+        index_radius_inner = int(vis_radius / tile_size)
+        index_radius_outer = index_radius_inner + 2
+
+        # the bounds for the visibility circle
+        lower_row = int(max(0, row - index_radius_outer))
+        lower_col = int(max(0, col - index_radius_outer))
+        upper_row = int(min(row + index_radius_outer, self.gridFull.num_rows))
+        upper_col = int(min(col + index_radius_outer, self.gridFull.num_cols))
+
+        lidar_data = self.generate_sensor.generateLidar(degree_freq, row, col)
+        rad_inc = int(GUI_tile_size / 3)  # radius increment to traverse tiles
+
+        def _color_normally(coords):
+            """
+            Colors the tile at coordinates based on its known attribute (obstacle, bloated, visited, or seen).
+            (coordinates must be in bounds of GUI window)
+            """
+            curr_tile = self.gridEmpty.grid[int(coords[0])][int(coords[1])]
+            curr_rec = self.tile_dict[curr_tile]
+            if curr_tile in self.visitedSet:  # tile on path already travelled
+                self.canvas.itemconfig(
+                    curr_rec, outline="#0C9F34", fill="#0C9F34")  # green
+            elif curr_tile.isBloated:
+                self.canvas.itemconfig(
+                    curr_rec, outline="#ffc0cb", fill="#ffc0cb")  # pink
+            elif curr_tile.isObstacle:
+                self.canvas.itemconfig(
+                    curr_rec, outline="#ff621f", fill="#ff621f")  # red
+            elif not curr_tile.isObstacle:
+                self.canvas.itemconfig(curr_rec, outline="#fff", fill="#fff")  # white
+                self.last_iter_seen.add(curr_rec)
+
+        for deg in range(0, 360, degree_freq):
+            # iterating through 360 degrees surroundings of robot in increments of degree_freq
+            angle_rad = deg * math.pi / 180
+            if len(lidar_data) == 0 or lidar_data[0][0] != deg:
+                # no object in sight at deg; color everything normally up to visibility radius
+                for r in range(0, index_radius_inner, rad_inc):
+                    coords = (r * math.sin(angle_rad) + row, r *
+                              math.cos(angle_rad) + col)  # (row, col) of tile
+                    if (coords[0] >= lower_row) and (coords[0] <= upper_row)\
+                            and (coords[1] >= lower_col) and (coords[1] <= upper_col):
+                        # make sure coords are in bounds of window
+                        _color_normally(coords)
+            else:  # obstacle in sight
+                # color everything normally UP TO obstacle, color obstacle red, then color unknown tiles gray
```

Comment:
It doesn't actually color unknow tiles gray right? It just leaves them as they were before???

Reply:
whoops inaccurate comment that should have been edited